remote-api.js

```diff
@@ -2,26 +2,15 @@
 const u = require('./util')
 const explain = require('explain-error')
 
-function isFunction (f) {
-  return typeof f === 'function'
-}
-
-function isObject (o) {
-  return o && typeof o === 'object'
-}
-
 // add all the api methods to the emitter recursively
 function recurse (obj, manifest, path, remoteCall) {
   for (const name in manifest) {
-    (function (name, type) {
-      const _path = path ? path.concat(name) : [name]
-      obj[name] =
-        isObject(type)
-          ? recurse({}, type, _path, remoteCall)
-          : function () {
-            return remoteCall(type, _path, [].slice.call(arguments))
-          }
-    })(name, manifest[name])
```

Comment:
I took a careful look at the git history, and I don't understand why was this block wrapped in an IIFE. I thought about it, and it seems to make sense to not have the IIFE inside the for loop